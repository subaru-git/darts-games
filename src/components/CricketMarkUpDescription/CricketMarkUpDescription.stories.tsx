/* eslint-disable import/named */
import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import CricketMarkUpDescription from './CricketMarkUpDescription';

export default {
  title: 'Components/CricketMarkUpDescription',
  component: CricketMarkUpDescription,
} as ComponentMeta<typeof CricketMarkUpDescription>;

const Template: ComponentStory<typeof CricketMarkUpDescription> = (args) => (
  <CricketMarkUpDescription {...args} />
);

export const Default = Template.bind({});
Default.args = {};
