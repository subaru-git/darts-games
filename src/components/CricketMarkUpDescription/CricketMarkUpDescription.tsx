import React, { FC } from 'react'
import { Button, Collapse, useDisclosure, Text, UnorderedList, ListItem } from '@chakra-ui/react'
import { GrCircleInformation } from 'react-icons/gr'
import LiteYoutubeEmbed from 'react-lite-youtube-embed'
import useLocale from '@/hooks/locale'
import 'react-lite-youtube-embed/dist/LiteYouTubeEmbed.css'

const CricketMarkUpDescription: FC = () => {
  const { isOpen, onToggle } = useDisclosure()
  const { t } = useLocale()
  return (
    <>
      <Button leftIcon={<GrCircleInformation />} variant='ghost' size='sm' onClick={onToggle}>
        What is this game?
      </Button>
      <Collapse in={isOpen}>
        <Text whiteSpace='pre-wrap'>{t.CRICKET_NUMBER_COUNT_DESCRIPTION}</Text>
        <UnorderedList py={2}>
          <ListItem>{t.CRICKET_NUMBER_COUNT_RULE_STEP1}</ListItem>
          <ListItem>{t.CRICKET_NUMBER_COUNT_RULE_STEP2}</ListItem>
        </UnorderedList>
        <Text>{t.CRICKET_NUMBER_COUNT_RULE_MORE}</Text>
        <LiteYoutubeEmbed
          id='RFl52NNvWhw'
          title='【ダーツ】菊地山口練習法　2週間でRt.4からRt.8になった練習法'
          webp
        />
      </Collapse>
    </>
  )
}

export default CricketMarkUpDescription
