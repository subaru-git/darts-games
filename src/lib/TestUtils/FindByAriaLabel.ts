export const findByAriaLabel = (elements: HTMLElement[], label: string) => {
  return (
    elements.find((button) => button.attributes.getNamedItem('aria-label')?.value === label) ??
    new HTMLElement()
  );
};
