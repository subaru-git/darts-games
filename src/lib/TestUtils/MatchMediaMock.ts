import mediaQuery from 'css-mediaquery'

export const createMatchMedia =
  (width: number) =>
  (query: string): MediaQueryList => ({
    matches: mediaQuery.match(query, { width }),
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })
