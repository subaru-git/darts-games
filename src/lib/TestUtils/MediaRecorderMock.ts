export const MediaRecorderMock = {
  start: jest.fn(),
  ondataavailable: jest.fn(),
  onerror: jest.fn(),
  state: '',
  stop: jest.fn(),
};
