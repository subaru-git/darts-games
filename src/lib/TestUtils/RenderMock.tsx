import { ReactNode } from 'react';
import { ChakraProvider } from '@chakra-ui/react';
import { render as defaultRender } from '@testing-library/react';
import Providers from '@/contexts/Providers';

const render = (children: ReactNode | ReactNode[]) =>
  defaultRender(
    <ChakraProvider>
      <Providers>{children}</Providers>
    </ChakraProvider>,
  );
export { render };
