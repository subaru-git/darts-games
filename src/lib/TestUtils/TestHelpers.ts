import { waitFor as originalWaitFor, waitForOptions } from '@testing-library/react';

const defaultTimeout = process.env.WAIT_FOR_DEFAULT_TIMEOUT ? parseInt(process.env.WAIT_FOR_DEFAULT_TIMEOUT, 10) : 5000;

export async function waitFor(callback: () => void,  options?: waitForOptions): Promise<void> {
  return originalWaitFor(callback, { timeout: defaultTimeout, ...options });
}