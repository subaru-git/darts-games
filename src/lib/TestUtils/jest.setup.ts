import mocksdk from 'firebase-mock';
import failOnConsole from 'jest-fail-on-console';
import { createMatchMedia } from '@/lib/TestUtils/MatchMediaMock';
import { MediaRecorderMock } from '@/lib/TestUtils/MediaRecorderMock';

jest.mock('next/router', () => ({ useRouter: jest.fn().mockReturnValue({ locale: 'en' }) }));
jest.mock('react-camera-pro');
jest.mock('@/firebase/client', () => ({
  analytics: mocksdk.analytics,
  db: mocksdk.firestore,
  storage: mocksdk.storage,
  auth: mocksdk.auth,
  functions: mocksdk.functions,
}));
jest.mock('@firebase/auth');
beforeAll(() => {
  window.matchMedia = createMatchMedia(1100);
  window.MediaRecorder = (jest.fn() as any).mockImplementation(() => MediaRecorderMock);
  global.ResizeObserver = require('resize-observer-polyfill');
});

failOnConsole();
