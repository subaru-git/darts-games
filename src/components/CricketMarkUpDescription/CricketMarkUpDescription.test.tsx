import { ChakraProvider } from '@chakra-ui/react';
import { render, screen } from '@testing-library/react';
import { createMatchMedia } from '../../lib/TestUtils/MatchMediaMock';
import CricketMarkUpDescription from './CricketMarkUpDescription';

jest.mock('next/router', () => ({ useRouter: jest.fn().mockReturnValue({ locale: 'en' }) }));
test('should rendering', () => {
  window.matchMedia = createMatchMedia(1100);
  const { container } = render(
    <ChakraProvider>
      <CricketMarkUpDescription />
    </ChakraProvider>,
  );
  expect(screen.getByText('What is this game?')).not.toBeNull();
  expect(container).toMatchSnapshot();
});
