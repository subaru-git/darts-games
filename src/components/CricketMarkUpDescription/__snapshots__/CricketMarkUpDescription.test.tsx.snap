// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendering 1`] = `
<div>
  <button
    class="chakra-button css-4xx2wk"
    type="button"
  >
    <span
      class="chakra-button__icon css-1wh2kri"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12,22 C17.5228475,22 22,17.5228475 22,12 C22,6.4771525 17.5228475,2 12,2 C6.4771525,2 2,6.4771525 2,12 C2,17.5228475 6.4771525,22 12,22 Z M12,10 L12,18 M12,6 L12,8"
          fill="none"
          stroke="#000"
          stroke-width="2"
        />
      </svg>
    </span>
    What is this game?
  </button>
  <div
    class="chakra-collapse"
    style="overflow: hidden; display: none; opacity: 0; height: 0px;"
  >
    <p
      class="chakra-text css-1kxrhf3"
    >
      This game was introduction by 'Kikuchi Yamaguchi.'
The game rule is:
    </p>
    <ul
      class="css-iq85eg"
      role="list"
    >
      <li
        class="css-0"
      >
        Setting target count (recommended 10 times)
      </li>
      <li
        class="css-0"
      >
        Shoot for each cricket numbers from 20 to 15 and Bull until the target count of times, like Cricket.
      </li>
    </ul>
    <p
      class="chakra-text css-0"
    >
      For more information, please watch this video.
    </p>
    <link
      as="image"
      href="https://i.ytimg.com/vi/RFl52NNvWhw/hqdefault.jpg"
      rel="preload"
    />
    <article
      class="yt-lite "
      data-title="【ダーツ】菊地山口練習法　2週間でRt.4からRt.8になった練習法"
      style="background-image: url(https://i.ytimg.com/vi/RFl52NNvWhw/hqdefault.jpg); --aspect-ratio: 56.25%;"
    >
      <button
        aria-label="Watch 【ダーツ】菊地山口練習法　2週間でRt.4からRt.8になった練習法"
        class="lty-playbtn"
      />
    </article>
  </div>
</div>
`;
